public/app/plugins/datasource/elasticsearch/datasource.ts

```diff
@@ -216,8 +226,18 @@ export class ElasticDatasource extends DataSourceApi<ElasticsearchQuery, Elastic
     const payload = angular.toJson(header) + '\n' + angular.toJson(data) + '\n';
 
     return this.post('_msearch', payload).then((res: any) => {
+      let parsedRes = undefined;
+      if (typeof res !== 'object') {
```

Comment:
I am not exactly sure what are we doing here. From the issue I thought that the problem was that elastic response wasn't json but we tried to parse it resulting in parsing error. Now we seem to actually add the parsing and then just again throw the parsing error. Not sure what is a purpose of that.